accepted/00000-coco-attestation.md

```diff
@@ -0,0 +1,245 @@
+It should be possible that one attestation container image can attest multiple environment types.
+But it may also happen that a new environment type require a new type of container image.
+
+
+## Database Design
+
+The table for the configuration should be a one-to-one relation to `rhnServer`.
+It has a boolean for attestation is enabled and an environment column for the environment of that system.
+
+```sql
+CREATE TABLE suseServerCoCoAttestationConfig
+(
+        server_id NUMERIC NOT NULL
+                    CONSTRAINT suse_srvcocoatt_cnf_sid_pk PRIMARY KEY,
+        CONSTRAINT suse_srvcocoatt_cnf_sid_fk FOREIGN KEY (server_id) REFERENCES rhnServer (id),
+        enabled   BOOLEAN NOT NULL DEFAULT FALSE,
+        environment VARCHAR(128) NULL
+                    CONSTRAINT suse_srvcocoatt_cnf_env_ck
+                    CHECK(environment IN ('kvm_amd_epyc_milan', 'kvm_amd_epyc_genoa'))
+);
+```
+
+The table for the report should be a many-to-one relation to hold a history of the attestations.
+The `data` column is of type JSONB to hold any kind of data in machine readable form collected from the
+VM to be attested.
+As different environments perform different checks, multiple data structures are expected.
+
+The status is calculated from the different results. It should be set to `succeeded` when all
+results have the status `succeeded` as well. As soon as one result is set to `failed` the
+overall status of the report should also show `failed`.
+
+```sql
+CREATE TABLE suseServerCoCoAttestationReport
+(
+        id          NUMERIC NOT NULL
+                      CONSTRAINT suse_srvcocoatt_rep_id_pk PRIMARY KEY,
+        server_id   NUMERIC NOT NULL
+                      CONSTRAINT suse_srvcocoatt_rep_sid_fk REFERENCES rhnServer (id),
+        environment VARCHAR(128) NULL
+                      CONSTRAINT suse_srvcocoatt_rep_env_ck
+                        CHECK(environment IN ('kvm_amd_epyc_milan', 'kvm_amd_epyc_genoa')),
+        status      VARCHAR(32) NOT NULL
+                      CONSTRAINT suse_srvcocoatt_rep_st_ck
+                        CHECK(status IN ('pending', 'succeeded', 'failed')),
+        data        JSONB NOT NULL, -- machine readable data
+        created     TIMESTAMPTZ DEFAULT (current_timestamp) NOT NULL,
+        modified    TIMESTAMPTZ DEFAULT (current_timestamp) NOT NULL
+);
+
+CREATE SEQUENCE suse_srvcocoatt_rep_id_seq;
+
+CREATE INDEX suse_srvcocoatt_rep_sid_idx
+  ON suseServerCoCoAttestationReport (server_id);
+
+CREATE INDEX suse_srvcocoatt_rep_stenv_idx
+  ON suseServerCoCoAttestationReport (status, environment);
+```
+
+The result table has the result for every check performed on the report.
+This is a many-to-one relation as many checks can be performed for one report.
+Every result has a type, status and summary which can be shown in a user interface.
+Optional it can have details if applicable.
+
+
+```sql
+CREATE TABLE suseServerCoCoAttestationResult
```

Comment:
The problem is, that only the attestation container knows what checks should be performed.
My idea is, that the java code does not know about all of this. In best case a new environment or check would just result in a change in the state and the container. No changes to Java.

Reply:
I will changed the RFC to have attestation containers for 1 or multiple result types instead of environments.
The mapping between environment types and required result types needs to be defined. Current suggestion is to define this in Java as it also need to create these entries.